.github/workflows/build_and_publish_main.yaml

```diff
@@ -1,10 +1,11 @@
-name: Main Validation and Release
+name: Main Validation and Release Workflow
 on:
-  push:
-    branches:
-      - main
+  pull_request_target:
+      types:
+        - closed
```

Comment:
Technically this runs on any PR that is merged - so even a feature branch off of dev being merged to dev. Is this what is intended? If so, change the name of the workflow and file to remove "main". If you want it to run only for main then maybe merge with the validate pull request workflow.

Reply:
added main